numpy/lib/_function_base_impl.py

```diff
@@ -4400,139 +4267,82 @@ def quantile(a,
-    since Python uses 0-based indexing, the code subtracts another 1 from the
-    index internally.
-
-    The following formula determines the virtual index ``i + g``, the location
-    of the quantile in the sorted sample:
-
-    .. math::
-        i + g = q * ( n - alpha - beta + 1 ) + alpha
+    By default, this is done by interpolating between adjacent elements in
+    ``y``, a sorted copy of `a`::
 
-    The different methods then work as follows
+        (1-g)*y[j] + g*y[j+1]
 
-    inverted_cdf:
-        method 1 of H&F [1]_.
-        This method gives discontinuous results:
+    where the index ``j`` and coefficient ``g`` are the integral and
+    fractional components of ``q * (n-1)``, and ``n`` is the number of
+    elements in the sample.
 
-        * if g > 0 ; then take j
-        * if g = 0 ; then take i
+    This is a special case of Equation 1 of H&F [1]_. More generally,
 
-    averaged_inverted_cdf:
-        method 2 of H&F [1]_.
-        This method gives discontinuous results:
+    - ``j = (q*n + m - 1) // 1``, and
+    - ``g = (q*n + m - 1) % 1``,
 
-        * if g > 0 ; then take j
-        * if g = 0 ; then average between bounds
+    where ``m`` may be defined according to several different conventions.
+    The preferred convention may be selected using the ``method`` parameter:
 
-    closest_observation:
-        method 3 of H&F [1]_.
-        This method gives discontinuous results:
+    =============================== =============== ===============
+    ``method``                      number in H&F   ``m``
+    =============================== =============== ===============
+    ``interpolated_inverted_cdf``   4               ``0``
+    ``hazen``                       5               ``1/2``
+    ``weibull``                     6               ``q``
+    ``linear`` (default)            7               ``1 - q``
+    ``median_unbiased``             8               ``q/3 + 1/3``
+    ``normal_unbiased``             9               ``q/4 + 3/8``
+    =============================== =============== ===============
 
-        * if g > 0 ; then take j
-        * if g = 0 and index is odd ; then take j
-        * if g = 0 and index is even ; then take i
+    Note that indices ``j`` and ``j + 1`` are clipped to the range ``0`` to
+    ``n - 1`` when the results of the formula would be outside the allowed
+    range of non-negative indices.
 
-    interpolated_inverted_cdf:
-        method 4 of H&F [1]_.
-        This method gives continuous results using:
+    The table above includes only the estimators from H&F that are continuous
+    functions of probability `q` (estimators 4-9). NumPy also provides the
+    three discontinuous estimators from H&F (estimators 1-3), where ``j`` is
+    defined as above and ``m`` and ``g`` are defined as follows.
 
-        * alpha = 0
-        * beta = 1
+    1. ``inverted_cdf``: ``m = 0`` and ``g = int(q*n > 0)``
```

Comment:
In my experience, this is the version used the most in statistical literature and corresponds to H&F's first equation in the introduction. But this equation is not general enough, it neglects distributions with flat regions, e.g. step functions.

Summary: better no action.

Reply:
I don't understand. Does "better no action" mean that there is no action to take to respond to this comment, or that I should not have taken action to adjust this part of the documentation?

Reply:
Yes, I suggest to leave it as is. Just wanted to share thoughts with a fellow statistian.